src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs

```diff
@@ -193,7 +512,149 @@ public AnnotationViewModel(WorkspaceViewModel workspaceViewModel, AnnotationMode
+            }
+
+            originalOutPorts = GetGroupOutPorts().Concat(groupedGroupsOutPorts);
+
+            List<PortViewModel> newPortViewModels = CreateProxyPorts(originalOutPorts);
+
+            if (newPortViewModels == null) return;
+            OutPorts.AddRange(newPortViewModels);
+        }
+
+        internal IEnumerable<PortModel> GetGroupInPorts(IEnumerable<NodeModel> ownerNodes = null)
+        {
+            // If this group does not contain any AnnotationModels
+            // we want to get all ports that are connected to something
+            // outside of the group
+            if (ownerNodes != null)
+            {
+                return Nodes.OfType<NodeModel>()
+                    .SelectMany(x => x.InPorts
+                        .Where(p => !p.IsConnected || !p.Connectors.Any(c => ownerNodes.Contains(c.Start.Owner)))
+                    );
+            }
+
+            // If this group does contain any AnnotationModels
+            // we only want to get the ports where the owner does
+            // not belong to a group.
+            return Nodes.OfType<NodeModel>()
+                .SelectMany(x => x.InPorts
+                    .Where(p => !p.IsConnected || !p.Connectors.Any(c => Nodes.Contains(c.Start.Owner)))
+                );
+        }
+
+        internal IEnumerable<PortModel> GetGroupOutPorts()
+        {
+            // If this group does not contain any AnnotationModels
+            // we want to get all ports that are connected to something
+            // outside of the group
+            if (!AnnotationModel.HasNestedGroups)
+            {
+                return Nodes.OfType<NodeModel>()
+                    .SelectMany(x => x.OutPorts
+                        .Where(p => !p.IsConnected || !p.Connectors.Any(c => Nodes.Contains(c.End.Owner)))
+                    );
+            }
+
+            // If this group does contain any AnnotationModels
+            // we only want to get the ports where the owner does
+            // not belong to a group.
+            return Nodes.OfType<NodeModel>()
+                .SelectMany(x => x.OutPorts
+                    .Where(p => !p.IsConnected || !p.Connectors.Any(c => c.End.Owner.BelongsToGroup))
+                );
+        }
+
+        private List<PortViewModel> CreateProxyPorts(IEnumerable<PortModel> groupPortModels)
+        {
+            var proxyModels = groupPortModels
+                .Select(x => new ProxyPortModel(x))
+                .ToList();
+
+            var originalPortViewModels = WorkspaceViewModel.Nodes
+                .SelectMany(x => x.InPorts.Concat(x.OutPorts))
+                .Where(x => groupPortModels.Contains(x.PortModel))
+                .ToList();
+
+            var newPortViewModels = new List<PortViewModel>();
+            for (int i = 0; i < proxyModels.Count(); i++)
+            {
+                var proxyModel = proxyModels[i];
+                //proxyModel.SetCenterPoint(new Point2D(0, 0));
```

Comment:
Is this still needed? I think from our discussion the position or the proxy does not work yet since the center point for port model is inaccurate?